packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -199,18 +205,23 @@ export class SummaryManager implements IDisposable {
                 return;
             }
 
+            // We transition to Running before requesting the summarizer, because after requesting we can't predict
+            // when the electedClient will be replaced with the new summarizer client.
+            // The alternative would be to let connectedState.clientId !== clientElection.electedClientId when
+            // state === Starting || state === Running.
+            assert(this.state === SummaryManagerState.Starting, 0x263 /* "Expected: starting" */);
```

Comment:
This assert moved to before requesting the summarizer. Do we expect to be in the starting state now? This seems a little risky so making sure.

Reply:
Yes, we expect it to be in starting state until the state is changed to Running. The real effect here is making the change to Running state earlier. That's to guarantee that until we're in Running state, _electedParent === _electedClient. After the summarizer client is spawned, _electedClient will be set to the summarizer client when the addClient message is processed.